template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidator.java

```diff
@@ -53,17 +55,17 @@ public void validateHostGroupScalingRequest(String accountId, Blueprint blueprin
         Set<String> services = templateProcessor.getComponentsByHostGroup().get(hostGroup.getName());
         for (BlackListedDownScaleRole role : BlackListedDownScaleRole.values()) {
             if (services.contains(role.name()) && adjustment < 0) {
-                validateRole(accountId, role, blueprintVersion);
+                validateRole(accountId, role, blueprintVersion, templateProcessor);
             }
         }
         for (BlackListedUpScaleRole role : BlackListedUpScaleRole.values()) {
             if (services.contains(role.name()) && adjustment > 0) {
-                validateRole(accountId, role, blueprintVersion);
+                validateRole(accountId, role, blueprintVersion, templateProcessor);
             }
         }
     }
 
-    private void validateRole(String accountId, EntitledForServiceScale role, Versioned blueprintVersion) {
+    private void validateRole(String accountId, EntitledForServiceScale role, Versioned blueprintVersion, CmTemplateProcessor templateProcessor) {
         boolean versionEnablesScaling = isVersionEnablesScaling(blueprintVersion, role);
         boolean entitledFor = entitlementService.isEntitledFor(accountId, role.getEntitledFor());
         if (role.getBlockedUntilCDPVersion().isPresent() && !versionEnablesScaling && !entitledFor) {
```

Comment:
Shouldn't this conditions be linked with OR instead of AND?

Reply:
I think this is right this way